### DISCUSSION TITLE
What affects the final output unit?

### DISCUSSION THREAD

**Discussion by moegodot** [harfbuzz/harfbuzz#3453] [Q&A] [answered]
I noticed that harfbuzz provide `hb_font_set_ptem` `hb_font_set_ppem` `hb_font_set_scale` `hb_face_set_upem` functions.

`hb_buffer_get_glyph_positions` always returns a particularly large number(such as 512 or 1024), regardless of the values set by `hb_font_set_ptem` and `hb_font_set_ppem`. 

I want to use pixels as the unit for the return value of `hb_buffer_get_glyph_positions`.

What should I do? Is there any documentation?

**Comment by behdad** [comment, marked as answer]
`hb_font_set_scale` is what affects the output scale of `hb_buffer_get_glyph_positions`. If you pass numbers like pixel font size to `hb_font_set_scale`, then you would get pixels out. But we highly recommend against doing that as you will get lots of rounding error and other artitfacs. We suggest using some subpixel precision like at least a few bitss.
